src/engine/cachingreader.h

```diff
@@ -151,12 +151,13 @@ class CachingReader : public QObject {
     // Gets a chunk from the free list, frees the LRU CachingReaderChunk if none available.
     CachingReaderChunkForOwner* allocateChunkExpireLRU(SINT chunkIndex);
 
-    enum class State {
-        Idle,
-        TrackLoading,
-        TrackLoaded,
+    enum State {
```

Comment:
Why was this changed from an enum class to an enum? Please change it back to an enum class.

Reply:
Because we need integers for the atomic. I didn't check if this still works with an `enum class`.

Reply:
So cast it to an int and back when needed.

Reply:
I actually had a version with a static_cast<int>() one every use. But @uklotzde version with a plain enum conviced me, because it is less distracting. In this case we have no namespace pollution issue, because the enum is class private.

Reply:
In this special case the enum literals are just private integer constants and a classical enum is a good fit.